Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -93,20 +121,28 @@ import Foundation
     @_documentation(visibility: private)
     public func fetchOrReturnRemoteConfiguration(_ completion: @escaping (BTConfiguration?, Error?) -> Void) {
         // TODO: - Consider updating all feature clients to use async version of this method?
-
-        Task { @MainActor in
-            do {
-                let configuration = try await configurationLoader.getConfig()
-                setupHTTPCredentials(configuration)
-                completion(configuration, nil)
-            } catch {
-                completion(nil, error)
+        
+        if authorization.type == .invalidAuthorization {
+            completion(nil, BTAPIClientError.invalidAuthorization(authorization.originalValue))
+        } else {
+            Task { @MainActor in
```

Comment:
Can you explain why we're dispatching to `MainActor` in our network layer? I feel like that should be up to the merchant when they get our async result from `tokenize()` with either an error or success

Reply:
This is not a change in code, it's just moving this logic into an else. But our code expects fetch or return config to be run on the main thread according to this test: https://github.com/braintree/braintree_ios/blob/258ae49bede22af809e229f9ba6486656955408c/UnitTests/BraintreeCoreTests/BTAPIClient_Tests.swift#L266